models/cam-fv/camdart_obs_preprocessor.rst

```diff
@@ -0,0 +1,122 @@
+program ``camdart_obs_preprocessor``
+====================
+
+Overview
+--------
+
+This program provides a way to modifiy the incoming observation stream before running filter
+for the CAM model.  Currently runs with the CAM-FV variant.  It uses the CAM model_mod namelist
+to select the CAM level above which to remove observations.  The output obs_seq file contains
+all observations below that level and none above.
+
+This program could be extended to superob dense obs, adjust obs error values near the surface,
+or any other desired changes to the incoming observation stream.
+
+|
+
+Usage
+-----
+
+Set the input and output obs_seq filenames in the namelist and run.
+
+|
+
+Namelist
+--------
+
+This namelist is read from the file ``input.nml``. Namelists start with an ampersand '&' and terminate with a slash '/'.
+Character strings that contain a '/' must be enclosed in quotes to prevent them from prematurely terminating the
+namelist.
+
+::
+
+
+   &camdart_obs_preprocessor_nml
+      filename_in  = 'obs_seq.out'
+      filename_out = 'obs_seq.nohighobs'
+      verbose      = .true.
+      calendar     = 'Gregorian'
+      print_every  = 5000
+      /
+
+| 
+
+Items in this namelist set the input and output files.
+
+.. container::
+
+   +--------------+--------------------+--------------------------------------------------------------------------------+
+   | Item         | Type               | Description                                                                    |
+   +==============+====================+================================================================================+
+   | filename_in  | character(len=256) | Observation sequence file to read                                              |
+   +--------------+--------------------+--------------------------------------------------------------------------------+
+   | filename_out | character(len=256) | Observation sequence file to create and write. If this file exists it will be  |
+   |              |                    | overwritten.                                                                   |
+   +--------------+--------------------+--------------------------------------------------------------------------------+
+   | verbose      | logical            | If true, write more debugging/status information.                              |
+   |              |                    |                                                                                |
+   +--------------+--------------------+--------------------------------------------------------------------------------+
+   | calendar     | character(len=32)  | The string name of a valid DART calendar type. (See the                        |
+   |              |                    | :doc:`../../modules/utilities/time_manager_mod` documentation for a list of    |
```

Comment:
links to other documentation look like ':doc:`../../../assimilation_code/'...
   in the github rendering.

Reply:
i'm going to leave the doc situation to others - i haven't done it enough to understand the current process, sorry.